pretyping/evarconv.ml

```diff
@@ -1751,8 +1740,7 @@ let solve_unconstrained_impossible_cases env evd =
   Evd.fold_undefined (fun evk ev_info evd' ->
     match ev_info.evar_source with
     | loc,Evar_kinds.ImpossibleCase ->
-      let j, ctx = coq_unit_judge env in
-      let evd' = Evd.merge_context_set Evd.univ_flexible_alg ?loc evd' ctx in
```

Comment:
This `_alg` was probably buggy, the cleanup discretely replaces it by univ_flexible (in `Evd.fresh_global`)

Reply:
Yep, it certainly was.